src/lib.rs

```diff
@@ -43,19 +46,6 @@ use indicatif::ProgressBar;
 pub struct Config {
     manifest_path: PathBuf,
     host: analysis::AnalysisHost,
-    assets: Vec<Asset>,
```

Comment:
While I like the changes overall and breaking things out I'd prefer this as is and not having it be a global static when it's really only used in one area. Perhaps a function to generate the assetfile when needed? We only call it in one place for the config and I think that's better than relying on global state.